### PR TITLE
Add Playwright binary mirror support

### PR DESCRIPTION
**Pull request by mnixry**
Fixes https://github.com/cnpm/cnpm/issues/359

### CODE REVIEW

File: app/common/adapter/binary/PlaywrightBinary.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            .catch(error =>
+              this.logger.warn(`Playwright version ${version} browser data request failed:`, error),
+            ),
+        ),
+      );
+
+      for (const [ version, browsers ] of browsersMap) {
+        for (const browser of browsers) {
+          const downloadPaths = DOWNLOAD_PATHS[browser.name];
+          if (!downloadPaths) return;
```
Comment (fengmk2) [inline app/common/adapter/binary/PlaywrightBinary.ts:167]: `continue` ?

File: app/common/adapter/binary/PlaywrightBinary.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+      this.dirItems = {
+        '/': [{ name: 'builds/', isDir: true, url: '', size: '-', date: nowDateISO }],
+        '/builds/': Object.keys(DOWNLOAD_PATHS).map(
+          dist => ({ name: `${dist}/`, isDir: true, url: '', size: '-', date: nowDateISO })),
+        ...Object.fromEntries(Object.keys(DOWNLOAD_PATHS).map(dist => [ `/builds/${dist}/`, []])),
+      };
+
+
+      // Only download beta and release versions of packages to reduce amount of request
+      const packageVersionsMap = new Map(Object.keys(packageData.versions)
```
Comment (fengmk2) [inline app/common/adapter/binary/PlaywrightBinary.ts:148]: 只取最新的 20 个 beta 和 release 版本？否则太多了。

**Comment by fengmk2** [comment]
@mnixry 参考 https://github.com/cnpm/cnpmcore/blob/main/test/common/adapter/binary/CypressBinary.test.ts#L18 写一下单测。

**Comment by fengmk2** [comment]
will be merge into https://github.com/cnpm/cnpmcore/pull/239

**Comment by mnixry** [comment]
Sorry，前几天在高考，没能抽出时间写😭还是麻烦大佬了

**Comment by fengmk2** [comment]
@mnixry 没事，打扰你高考了，上了大学之后继续来。

**Comment by fengmk2** [comment]
已经在同步 https://cnpmjs.org/mirrors/playwright/builds/chromium/1000/chromium-linux.zip

**Comment by fengmk2** [comment]
https://r.cnpmjs.org/binary.html?path=playwright/builds/chromium/1000/

**Comment by fengmk2** [comment]
https://registry.npmmirror.com/binary.html?path=playwright/builds/chromium-tip-of-tree/1005/

**Pull request closed without merging**